src/Mappers/Root/EnumTypeMapper.php

```diff
@@ -0,0 +1,199 @@
+ */
+class EnumTypeMapper implements RootTypeMapperInterface
+{
+    /** @var array<class-string<UnitEnum>, EnumType> */
+    private $cache = [];
+    /** @var array<string, EnumType> */
+    private $cacheByName = [];
+    /** @var RootTypeMapperInterface */
+    private $next;
+    /** @var AnnotationReader */
+    private $annotationReader;
+    /** @var array|NS[] */
+    private $namespaces;
+    /** @var CacheInterface */
+    private $cacheService;
+
+    /**
+     * @param NS[] $namespaces List of namespaces containing enums. Used when searching an enum by name.
+     */
+    public function __construct(RootTypeMapperInterface $next, AnnotationReader $annotationReader, CacheInterface $cacheService, array $namespaces)
+    {
+        $this->next = $next;
+        $this->annotationReader = $annotationReader;
+        $this->cacheService = $cacheService;
+        $this->namespaces = $namespaces;
+    }
+
+    /**
+     * @param (OutputType&GraphQLType)|null $subType
+     * @param ReflectionMethod|ReflectionProperty $reflector
+     *
+     * @return OutputType&GraphQLType
+     */
+    public function toGraphQLOutputType(Type $type, ?OutputType $subType, $reflector, DocBlock $docBlockObj): OutputType
+    {
+        $result = $this->map($type);
+        if ($result === null) {
+            return $this->next->toGraphQLOutputType($type, $subType, $reflector, $docBlockObj);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param (InputType&GraphQLType)|null $subType
+     * @param ReflectionMethod|ReflectionProperty $reflector
+     *
+     * @return InputType&GraphQLType
+     */
+    public function toGraphQLInputType(Type $type, ?InputType $subType, string $argumentName, $reflector, DocBlock $docBlockObj): InputType
+    {
+        $result = $this->map($type);
+        if ($result === null) {
+            return $this->next->toGraphQLInputType($type, $subType, $argumentName, $reflector, $docBlockObj);
+        }
+
+        return $result;
+    }
+
+    private function map(Type $type): ?EnumType
+    {
+        if (! $type instanceof Object_) {
+            return null;
+        }
+        $fqsen = $type->getFqsen();
+        if ($fqsen === null) {
+            return null;
+        }
+
+        /** @var class-string<object> $enumClass */
+        $enumClass = (string) $fqsen;
+
+        return $this->mapByClassName($enumClass);
+    }
+
+    /**
+     * @param class-string $enumClass
+     */
+    private function mapByClassName(string $enumClass): ?EnumType
+    {
+        if (isset($this->cache[$enumClass])) {
+            return $this->cache[$enumClass];
+        }
+
+        if (! enum_exists($enumClass)) {
+            return null;
+        }
+
+        // phpcs:disable SlevomatCodingStandard.Commenting.InlineDocCommentDeclaration.MissingVariable
+        /** @var class-string<UnitEnum> $enumClass */
+        // phpcs:enable SlevomatCodingStandard.Commenting.InlineDocCommentDeclaration.MissingVariable
```

Comment:
Are we getting much value out of this?  I'd just remove it.

Reply:
PHPStan (before I disabled it) required a `class-string<UnitEnum>`, as signed in `EnumType` constructor, and wasn't able to determine `$enumClass` complies at this point (after the `enum_exist` check).

On the other side, CodeSniffer doesn't like precising the type subsequently to a variable declaration (which I think is a shame, but I guess I understand the rule), therefore the `phpcs:disable` directive.

I could simply soften the typecheck in `EnumType::__constructor`, from `class-string<UnitEnum>` to `class-string`, but I must say I'm not so fond of this option.

Reply:
IMO, we should be doing better validation in the constructor for this string, instead of relying on static type analysis.  What kind of feedback is a developer going to here here if an incorrect class-string is provided?  It's nice to provide these additional static checks for IDE completion and lib development testing, but provides little runtime, or even development, assurances.

Is this error not referring to the fact that it's a `param` and not a `var` in terms of phpcs parsing?  Why aren't we applying this in the docblock?